### PR TITLE
Bump React to 18, bump deps

### LINKED ISSUE #49: React 18 upgrade
Cannot proceed until https://github.com/chakra-ui/chakra-ui/issues/5844 is merged.

### PR DESCRIPTION
**Pull request by IAmVisco**
Closes #49.

### CODE REVIEW

File: src/components/channel/ChannelCard.tsx [hunk 1]
```diff
@@ -40,11 +40,11 @@ export function ChannelCard({ channel, ...rest }: ChannelCardProps) {
         boxShadow="0 8px 24px rgba(0,0,0,0.5)"
       />
       <Text
-        isTruncated
```
Comment (IAmVisco) [inline src/components/channel/ChannelCard.tsx:43]: https://chakra-ui.com/getting-started/migration#styled-system

File: src/components/data/SongTable/UpcomingSongList.tsx [hunk 2]
```diff
@@ -76,7 +76,7 @@ const QueueHeader = () => {
     <Flex mt={4} gap={2} alignItems="center">
       <Text fontSize={["md", "lg"]}>
         <Text opacity={0.66} as={"span"}>
-          {t("Queue")}
+          {t("Queue") as string}
```
Comment (IAmVisco) [inline src/components/data/SongTable/UpcomingSongList.tsx:79]: react-i18next is still not ready for React 18 (but they claim it is https://github.com/i18next/react-i18next/issues/1476)

File: src/components/data/SongTable/UpcomingSongList.tsx [hunk 3]
```diff
@@ -176,17 +175,21 @@ export const UpcomingSongList = ({
...
+  const onScroll = useCallback(
+    ({ scrollTop }: { scrollLeft: number; scrollTop: number }) => {
+      list.current?.scrollTo(scrollTop);
+    },
+    []
+  );
 
   const renderList = useCallback(
     ({ height, width }: { height: number; width: number }) => (
+      // @ts-ignore React 18 typings issue
```
Comment (IAmVisco) [inline src/components/data/SongTable/UpcomingSongList.tsx:192]: react-virtualized wasn't updated for quite some time, no support for React 18 typings

File: src/components/layout/Frame.tsx [hunk 4]
```diff
@@ -124,8 +124,10 @@ export default function Frame({ children }: { children?: ReactNode }) {
   }, [currentlyPlaying.song, currentlyPlaying.repeat, loadPlayer]);
 
   function onReady(event: { target: YouTubePlayer }) {
-    setPlayer(event.target);
+    // @ts-ignore Hack to not pass `player` instance without mounted iframe
+    event.target.i && setPlayer(event.target);
```
Comment (IAmVisco) [inline src/components/layout/Frame.tsx:128]: This cost me a whole evening.

File: src/components/player/Player.tsx [hunk 5]
```diff
@@ -230,54 +222,52 @@ export function Player({ player }: { player: any }) {
...
-      next({ count: 1, userSkipped: false, hasError: true });
-      setError(false);
+    const MAX_RETRIES = 3;
+    if (!(player && hasError && currentSong)) return;
+
+    if (!retryCounts[currentSong.video_id])
+      retryCounts[currentSong.video_id] = 0;
+
+    if (retryCounts[currentSong.video_id] < MAX_RETRIES) {
+      // if (getID(player.getVideoUrl()) !== currentSong.video_id) return;
```
Comment (IAmVisco) [inline src/components/player/Player.tsx:234]: Not sure how to do async sequential timeouts inside of useEffect so just commented out that line for now.

File: src/components/player/YoutubePlayer.tsx [hunk 6]
```diff
@@ -1,22 +1,20 @@
...
   onReady,
 }: {
-  onReady: (event: { target: YouTubePlayer }) => void;
+  onReady: YouTubeProps["onReady"];
 }) {
   return (
     <YouTube
-      containerClassName="yt-player"
       className="yt-player"
-      videoId={""}
```
Comment (IAmVisco) [inline src/components/player/YoutubePlayer.tsx:14]: This line was doing funky stuff with iframe, removed.

File: src/index.tsx [hunk 7]
```diff
@@ -33,10 +33,14 @@ console.log(
 );
 (window as any)["App_Version"] = process.env.REACT_APP_VERSION;
 
+type Props = StoreProvider["props"] & { children: React.ReactNode };
```
Comment (IAmVisco) [inline src/index.tsx:36]: https://github.com/ctrlplusb/easy-peasy/issues/741

File: src/pages/Video.tsx [hunk 8]
```diff
@@ -67,15 +67,17 @@ export default function Video() {
             }}
             mb={2}
           >
+            {/* @ts-ignore */}
```
Comment (IAmVisco) [inline src/pages/Video.tsx:70]: This thing starts throwing weird typing issues, not even sure which lib is acting weird...

File: src/components/data/SongTable/index.tsx [hunk 9]
```diff
@@ -69,15 +69,20 @@ export const SongTable = ({
...
+  const onScroll = useCallback(
+    ({ scrollTop }: { scrollLeft: number; scrollTop: number }) => {
+      list.current?.scrollTo(scrollTop);
+    },
+    []
+  );
   if (!songList) return <>{t("No Songs")}</>;
 
   return virtualized ? (
+    // @ts-ignore
```
Comment (sphinxrave) [inline src/components/data/SongTable/index.tsx:81]: what's with these ts ignores :o

File: src/components/data/SongTable/UpcomingSongList.tsx (hunk 3 above)
Comment (IAmVisco) [reply src/components/data/SongTable/UpcomingSongList.tsx:192]: 🤔 https://github.com/DaPulse/react-virtualized

File: src/components/player/Player.tsx (hunk 5 above)
Comment (uetchy) [reply src/components/player/Player.tsx:237]: Can you add a `NOTE:` comment in the code as well for future reference?

**Review by uetchy — APPROVED** [review]

**Pull request merged**